### PR TITLE
Implements the update documents by function

### PR DESCRIPTION
**Pull request by irevoire**
# Pull Request

## Related issue
Fixes https://github.com/meilisearch/meilisearch-php/issues/661

## What does this PR do?
- Implements the update documents by function method
- Add a test

### CODE REVIEW

File: src/Endpoints/Delegates/HandlesDocuments.php [hunk 1]
```diff
@@ -139,6 +139,21 @@ public function updateDocumentsNdjsonInBatches(string $documents, ?int $batchSiz
...
     }
 
+    /**
+     * This is an EXPERIMENTAL feature, which may break without a major version.
+     * It's available after Meilisearch v1.10.
+     *
+     * More info about the feature: https://github.com/orgs/meilisearch/discussions/762
+     * More info about experimental features in general: https://www.meilisearch.com/docs/reference/api/experimental-features
+     *
+     * @param non-empty-string @function
```
Comment (norkunas) [inline src/Endpoints/Delegates/HandlesDocuments.php:149]:
```suggestion
     * @param non-empty-string $function
```

File: src/Endpoints/Delegates/HandlesDocuments.php [hunk 2]
```diff
@@ -139,6 +139,21 @@ public function updateDocumentsNdjsonInBatches(string $documents, ?int $batchSiz
...
+     * @param array{filter: non-empty-string|list<non-empty-string>, context: array}|null @options
```
Comment (norkunas) [inline src/Endpoints/Delegates/HandlesDocuments.php:150]:
```suggestion
     * @param array{filter: non-empty-string|list<non-empty-string>, context: array<non-empty-string, mixed>}|null $options
```

**Review by norkunas — APPROVED** [review]

File: src/Endpoints/Delegates/HandlesDocuments.php [hunk 3]
```diff
@@ -139,6 +139,21 @@ public function updateDocumentsNdjsonInBatches(string $documents, ?int $batchSiz
...
+     *
+     * More info about the feature: https://github.com/orgs/meilisearch/discussions/762
+     * More info about experimental features in general: https://www.meilisearch.com/docs/reference/api/experimental-features
+     *
+     * @param non-empty-string $function
+     * @param array{filter: non-empty-string|list<non-empty-string>|null, context: array<non-empty-string, mixed>}|null $options
+     */
+    public function updateDocumentsByFunction(string $function, ?array $options)
+    {
+        return $this->http->post(self::PATH.'/'.$this->uid.'/documents/edit', ['function' => $function, ...$options]);
```
Comment (norkunas) [inline src/Endpoints/Delegates/HandlesDocuments.php:154]:
```suggestion
     * @param array{filter: non-empty-string|list<non-empty-string>|null, context: array<non-empty-string, mixed>} $options
     */
    public function updateDocumentsByFunction(string $function, array $options = [])
    {
        return $this->http->post(self::PATH.'/'.$this->uid.'/documents/edit', ['function' => $function, ...$options]);
```
spread operator, cannot work with non array

File: src/Endpoints/Delegates/HandlesDocuments.php [hunk 4]
```diff
@@ -139,6 +139,21 @@ public function updateDocumentsNdjsonInBatches(string $documents, ?int $batchSiz
...
 
+    /**
+     * This is an EXPERIMENTAL feature, which may break without a major version.
+     * It's available after Meilisearch v1.10.
+     *
+     * More info about the feature: https://github.com/orgs/meilisearch/discussions/762
+     * More info about experimental features in general: https://www.meilisearch.com/docs/reference/api/experimental-features
+     *
+     * @param non-empty-string $function
+     * @param array{filter: non-empty-string|list<non-empty-string>|null, context: array<non-empty-string, mixed>} $options
```
Comment (norkunas) [inline src/Endpoints/Delegates/HandlesDocuments.php:150]:
```suggestion
     * @param array{filter?: non-empty-string|list<non-empty-string>|null, context?: array<non-empty-string, mixed>} $options
```
If filter or context is not mandatory

**Comment by curquiza** [comment]
@irevoire I cannot merge despite the approval of @norkunas because the tests are failing 😊

**Review by curquiza — DISMISSED** [review]
@irevoire can you update the tests? 😇

**Review by curquiza — CHANGES_REQUESTED** [review]
Request of change for the tests (no miss click this time 🤞 )

**Review by curquiza — APPROVED** [review]

**Pull request merged**